MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
 }
 
 //------------------------------------------------------------------------------
+void vtkSlicerMarkupsToModelLogic::ComputePointParametersRawIndices(vtkPolyData * markupsPointsPolyData, vtkDoubleArray* markupsPointsParameters)
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  if (markupsPointsParameters->GetNumberOfTuples())
+  {
+    // this should never happen, but in case it does, output a warning
+    vtkWarningMacro("markupsPointsParameters already has contents. Clearing.");
+    while (markupsPointsParameters->GetNumberOfTuples()) // clear contents just in case
+      markupsPointsParameters->RemoveLastTuple();
+  }
+
+  for (int v = 0; v < numPoints; v++)
+  {
+    markupsPointsParameters->InsertNextTuple1( v / double(numPoints-1) );
+    // division to clamp all values to range 0.0 - 1.0
+  }
+}
+
+//------------------------------------------------------------------------------
```

Comment:
I agree that the class is big, and I think there are a lot of reasons for this. I think refactoring the logic should be done as its own separate task, that way we can properly plan which parts belong in the class, and which parts should be in other files. We can also set up the proper vtk pipeline in this task to make the module overall more efficient.